config/cmake/HDFCXXCompilerFlags.cmake

```diff
@@ -134,11 +134,11 @@ else ()
         AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 4.8)
       # add the general CXX flags for g++ compiler versions 4.8 and above.
       ADD_H5_FLAGS (HDF5_CMAKE_CXX_FLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/cxx-general")
-      if (HDF5_ENABLE_WARNINGS_AS_ERRORS)
```

Comment:
Just modify the Add_H5_FLAGS macro to check the HDF5_ENABLE_WARNINGS_AS_ERRORS option.
This would clean up the if else tree here. (Maybe even move the CMAKE_[C,CXX]_COMPILER_ID if in there also?)